app/Traits/HasPointsAttribute.php

```diff
@@ -65,4 +71,19 @@ public function setPointsAttribute(array $points)
 
         $this->attributes['points'] = json_encode($points);
     }
+
+    /**
+     * Counts number of distinct points
+     * @param array $points containing the coordinates
+     * @return int number of distinct points
+     * **/
+    private function countDistinctCoordinates($points)
+    {
+        $points = collect($points);
+        // Use values to reset index
+        $x = $points->filter(fn ($x, $idx) => $idx % 2 === 0)->values();
+        $y = $points->filter(fn ($x, $idx) => $idx % 2 === 1)->values();
+        $coords = collect($x->map(fn ($x, $idx) => [$x, $y[$idx]]))->unique();
```

Comment:
This could also be a perfect use case for the [zip()](https://laravel.com/docs/10.x/collections#method-zip) method.
```suggestion
        $coords = $x->zip($y)->unique();
```